plugins/producer_plugin/producer_plugin.cpp

```diff
@@ -2011,6 +2040,8 @@ producer_plugin_impl::start_block_result producer_plugin_impl::start_block() {
             process_scheduled_and_incoming_trxs( scheduled_trx_deadline, incoming_itr );
          }
 
+         process_read_only_transactions( preprocess_deadline );
```

Comment:
I don't understand. Line #s have shifted since you wrote this comment, but I do not see us canceling processing ro transactions when building a block.

Reply:
Oh, my bad, I read the check incorrectly:

![image](https://user-images.githubusercontent.com/1496907/228070860-3f618706-4812-44c1-baec-fa0b7a8d168c.png)

I thought we were returning when `chain.is_building_block()` (missed the `!`).

But I'm a bit puzzled. I thought ro transactions did not go into blocks?

Reply:
ro trx do not go into blocks. However, we require a pending block for the execution of read-only transactions. This provides host function capability such as `get_block_num()`